### ISSUE TITLE
Support end early/start late: user setting to round appointments to nearest half hours

### ISSUE THREAD

**Issue by okms** [Puzzlepart/did#1056]
- [x] I have searched for existing issues to avoid creating duplicates

### TLDR
As part of their initiative to promote mental health and give workers an opportunity to breathe between Teams meetings, Microsoft introduced the option to allow people to end meetings early or start them late. This is an excellent feature which encourages us to get up, go to the bathroom etc between meetings. Did does however count the minutes as written, meaning a 25-minute meeting will count as 25 minutes logged. With manual time tracking, this appointment would be counted as 30 minutes. Likewise, a meeting lasting 50 minutes or 1h 50 m should be counted as 1 hour or 2 hours, respectively. 

We are currently doing workarounds to not "loose" the missing minutes in these appointments, like ignoring the original event and creating shadow events which we register in Did. This is tedious and is counter to the mission statement for Did. 

Those who are not doing these workarounds can lose up to several hours of "unlogged" time per week

### Description

In outlook we have the following options when shortening duration for events
- Events can end early or start late
  - eg: what would be a 1100-1130 meeting can be 1100-1125 or 1105-1130 
- Meetings less than an hour can be shortened by 0, 5, 8 or 10 minutes
- Meetings longer than an hour can be shortened by 0, 5, 10 or 15 minutes

The most common configuration is to have meetings less than an hour end 5 minutes before, and longer than 1 hour end 10 minutes before. 

I suggest the following configuration

- User setting "Round events to nearest 30 minutes" which is default off for all users
- If an event ends on the quarter (XX:00, XX:15, XX:30, XX:45) they will not be rounded
- for all other events they will be rounded to the nearest 30 minutes
  - A meeting from 11:30 to 11:55 will count as 30 minutes and be logged as 11:30 to 12:00
  - A meeting from 11:30 to 12:20 will count as 1 hour and be logged as 11:30 to 12:30
  - A meeting from 12:05 to 13:50 will count as 2 hours and be logged as 12:00 to 14:00
  - A meeting from 12:15 to 12:45 will not be affected by this feature

In the Timesheet view, appointments affeced (rounded to the nearest 30 minutes) get an icon showing it's been rounded due to user settings. 

example from @olemp 

![CleanShot 2022-09-30 at 17 21 01](https://user-images.githubusercontent.com/170147/193303056-b80c4966-895c-4dbb-85c2-ba9942ccb1c8.png)
